### DISCUSSION TITLE
ESP32 Port  build fails

### DISCUSSION THREAD

**Discussion by krishnak** [micropython/micropython#9641] [Core Development]
I am trying to add a single/simple function to esp32/modnetwork.c which in turn calls a function in #include "lwip/lwip_napt.h"

I have added the include declaration in the C file, yet it complains, What am I missing?

```
[ 83%] Generating ../../frozen_content.c
/home/krishnak/micropython/ports/esp32/modnetwork.c: In function 'esp_nat_enable':
/home/krishnak/micropython/ports/esp32/modnetwork.c:139:5: error: implicit declaration of function 'ip_napt_enable'; did you mean 'esp_nat_enable'? [-Werror=implicit-function-declaration]
     ip_napt_enable(htonl(napt_netif_ip), 1);
     ^~~~~~~~~~~~~~
     esp_nat_enable
cc1: some warnings being treated as errors



```

**Comment by robert-hh** [comment]
Dies the file lwip_napt.h exist? And if yes, in one of the include paths? I do not have it in my repository.
I find it here at `esp-idf/components/lwip/lwip/src/include`

**Comment by krishnak** [comment]
It exists exactly where you point out, the dns.h which is referred in the original code of modenetwork.c also resides there, so if it can find the dns.h it should also find this one?

**Reply by robert-hh** [reply]
Looking into the file, there are these three #if before the declaration of the function. Did you enable them all in your settings?
```
#if ESP_LWIP
#if IP_FORWARD
#if IP_NAPT
```

**Reply by krishnak** [reply]
I have 
CONFIG_LWIP_IP_FORWARD=y
CONFIG_LWIP_IPV4_NAPT=y

in the sdkconfig inside esp32/build

**Reply by krishnak** [reply]
A standalone C code compiles with the above settings, I am having issues only with the micropython integration

**Reply by krishnak** [reply]
OK Resolved, those config parameters need to be present in sdkconfig.base under /esp32/boards, it was incorrectly added to /esp32/build/sdkconfig
